src/main/java/org/araymond/joal/core/ttorrent/client/Client.java

```diff
@@ -196,10 +196,15 @@ public void onTorrentHasStopped(final Announcer stoppedAnnouncer) {
     @Override
     public void onTorrentFileAdded(final MockedTorrent torrent) {
         this.eventPublisher.publishEvent(new TorrentFileAddedEvent(torrent));
-        if (!this.stop && this.currentlySeedingAnnouncers.size() < this.appConfig.getSimultaneousSeed()) {
-            Lock lock = this.lock.writeLock();
+
+        if (this.stop) {
+            return;
+        }
+
+        Lock lock = this.lock.writeLock();
+        lock.lock();
+        if (this.currentlySeedingAnnouncers.size() < this.appConfig.getSimultaneousSeed()) {
```

Comment:
Reverting back to how it was -- this way `        if (this.currentlySeedingAnnouncers.size()` is backed by a lock.